model/schema/highThroughputExpression.yaml

```diff
@@ -0,0 +1,578 @@
+                expression_assay_used:
+                  required: true
+                sample_type:
+                  required: true
+          description: >-
+            At least sample_id, dataset_ids, expression_assay_used, and sample_type should be populated OR
+            at least sample_title, dataset_ids, expression_assay_used, and sample_type should be populated
+
+  HighThroughputExpressionDatasetSampleAnnotationDTO:
+    is_a: SubmittedObjectDTO
+    description: >-
+     An entry with dataset sample object associations.
+    slots:
+      - sample_id_dto # sampleId in agr_schemas json
+      - sample_title # sampleTitle in agr_schemas json
+      - sample_type_curie # sampleType in agr_schemas json
+      - sample_age_dto # sampleAge in agr_schemas json
+      - sample_location_dtos # sampleLocations in agr_schemas json
+      - where_expressed_statement # whereExpressedStatement in whereExpressed.json
+      - abundance # abundance in agr_schemas json
+      - genomic_information # genomicInformation in agr_schemas json
+      - taxon # taxonId in agr_schemas json
+      - genetic_sex # sex in agr_schemas json
+      - expression_assay_curie # assayType in agr_schemas json
+      - sequencing_format # sequencingFormat in agr_schemas json
+      - assembly_versions # assemblyVersions in agr_schemas json
+      - note_dtos # notes in agr_schemas json
+      - dataset_ids # datasetIds in agr_schemas json
+      - microarray_sample_details # microarraySampleDetails in agr_schemas json
+    slot_usage:
+      sample_id: # TO DO update this section
+        description: >-
+          The ID for this sample object, MOD or external database ID (ex: GEO);
+          primaryId and secondaryId
+      genetic_sex:
+        description: >-
+          The sex of the organism the sample came from
+      expression_assay_used:
+        description: >-
+          ID of the type of assay used for this sample; sample or experiment property? See enumeration below.
+        range: high_throughput_expression_assay_values
+    rules:
+      - postconditions:
+          any_of:
+            - slot_conditions:
+                sample_id:
+                  required: true
+                dataset_ids:
+                  required: true
+                expression_assay_used:
+                  required: true
+                sample_type_curie:
+                  required: true
+            - slot_conditions:
+                sample_title:
+                  required: true
+                dataset_ids:
+                  required: true
+                expression_assay_used:
+                  required: true
+                sample_type_curie:
+                  required: true
+          description: >-
+            At least sample_id, dataset_ids, expression_assay_used, and sample_type should be populated OR
+            at least sample_title, dataset_ids, expression_assay_used, and sample_type should be populated
+
+  HighThroughputExpressionDatasetID:
```

Comment:
Currently this is only used for `sample_id` in `HighThroughputExpressionDatasetSampleAnnotation`.  If this is not intended to be used for `dataset_id` then can it be written as a mixin or just incorporated into the class itself to avoid unnecessary join tables?

Reply:
Maybe? I'd have to think on why there was a separate class created in the old JSON and whether having a separate class can be avoided. I've reworked the dataset_id slot. I may have missed your point

Reply:
My point is moot given that this class is now being used for dataset_id and sample_id.  Having said that, I'm not sure that the modelling of this is the best way forward.
To me it seems like a HighThroughputExpressionDatasetAnnotation needs an ID field and cross-references (if we want to indicate a preferred crossreference then that could be achieved by an additional boolean slot on the CrossReference class).
Similary the HighThroughputExpressionDatasetSampleAnnotation needs an ID field and cross-references.  
Then we need to link the dataset samples to the datasets.  In the old schema this is done via the datasetIDs field on the sample entry but in LinkML this could be modelled as either an association, a slot that is a list of annotations on the sample class, or a slot that is a list of samples on the annotation class.
This would remove the need for this separate class (which if we were to keep should be an AuditedObject not a SubmittedObject) as the ID would be covered by modEntityId/modInternalId and each class would just have a cross_references slot.